platforms/documentation/docs/src/docs/userguide/best-practices/best_practices_general.adoc

```diff
@@ -175,3 +175,118 @@ include::sample[dir="snippets/bestPractices/avoidInternal/groovy",files="build.g
 === Tags
 
 <<tags_reference.adoc#tag:upgrades,`#upgrades`>>
+
+[[no_source_in_root]]
+== Do Not Put Source Files in the Root Project
+:keywords: structuring-builds
+
+Do not put source files in your root project; instead, put them in an included (sub)project.
```

Comment:
(sub)project parentheses seem unnecessary. Can we just use subproject or sub-project?

Reply:
Yeah, I was trying to be too fancy here.  I'll change this to consistently say just "projects".